platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -247,33 +252,38 @@ static void mbed_minimal_formatted_string_void_pointer(char *buffer, size_t leng
     mbed_minimal_putchar(buffer, length, result, 'x', stream);
 
     /* write rest as a regular hexadecimal number */
-    mbed_minimal_formatted_string_hexadecimal(buffer, length, result, (ptrdiff_t) value, stream, true);
+    mbed_minimal_formatted_string_integer(buffer, length, result, (ptrdiff_t) value, HEX_UPPER, 0, false, stream);
 }
 
 #if MBED_CONF_PLATFORM_MINIMAL_PRINTF_ENABLE_FLOATING_POINT
 /**
  * @brief      Write double.
  *
- * @param      buffer  The buffer to store output (NULL for stdout).
- * @param[in]  length  The length of the buffer.
- * @param      result  The current output location.
- * @param[in]  value   The value to be printed.
+ * @param      buffer          The buffer to store output (NULL for stdout).
+ * @param[in]  length          The length of the buffer.
+ * @param      result          The current output location.
+ * @param[in]  value           The value to be printed.
+ * @param[in]  dec_precision   The decimal precision. If PRECISION_DEFAULT MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS is used.
  */
-static void mbed_minimal_formatted_string_double(char *buffer, size_t length, int *result, double value, FILE *stream)
+static void mbed_minimal_formatted_string_double(char *buffer, size_t length, int *result, double value, int dec_precision, FILE *stream)
 {
     /* get integer part */
     MBED_SIGNED_STORAGE integer = value;
 
     /* write integer part */
-    mbed_minimal_formatted_string_signed(buffer, length, result, integer, stream);
+    mbed_minimal_formatted_string_integer(buffer, length, result, integer, INT_SIGNED, 0, false, stream);
 
     /* write decimal point */
     mbed_minimal_putchar(buffer, length, result, '.', stream);
 
     /* get decimal part */
     double precision = 1.0;
 
-    for (size_t index = 0; index < MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS; index++) {
+    if (dec_precision == PRECISION_DEFAULT) {
```

Comment:
You need to handle a decimal precision of 0 right from the start. You can pass `width_size` and `prepend_zeros` to `mbed_minimal_formatted_string_integer` once you have adjusted for the decimal precision.

```
static void mbed_minimal_formatted_string_double(char *buffer, size_t length, int *result, double value, int dec_precision, int width_size, bool prepend_zeros, FILE *stream)
{
    /* get integer part */
    MBED_SIGNED_STORAGE integer = value;

    if (dec_precision == PRECISION_DEFAULT) {
        dec_precision = MBED_CONF_PLATFORM_MINIMAL_PRINTF_SET_FLOATING_POINT_MAX_DECIMALS;
    }

    if (dec_precision != 0) {
        width_size -= dec_precision + 1; // decimal precision plus '.'
        if (width_size < 0) width_size = 0;
    } else {
        value = (value - integer) * 1.0;
        if (!((value > -0.5) && (value < 0.5))) {
            integer++;
        }
    }

    /* write integer part */
    mbed_minimal_formatted_string_integer(buffer, length, result, integer, INT_SIGNED, width_size, prepend_zeros, stream);

    if (dec_precision != 0) {
        /* write decimal point */
        mbed_minimal_putchar(buffer, length, result, '.', stream);
       [...]
   }
}
```

Reply:
Ok Done